src/Library/demos/Action Bar/code.rs

```diff
@@ -0,0 +1,21 @@
+use crate::workbench;
+use gtk::prelude::*;
+
+pub fn main() {
+    let action_bar: gtk::ActionBar = workbench::builder().object("action_bar").unwrap();
+    let button: gtk::ToggleButton = workbench::builder().object("button").unwrap();
+    let start_widget: gtk::Button = workbench::builder().object("start_widget").unwrap();
+    let end_widget: gtk::Button = workbench::builder().object("end_widget").unwrap();
+
+   button.connect_notify_local(Some("active"), move |button, _| {
+    action_bar.set_revealed(!button.is_active());
+});
```

Comment:
Please make sure to always run the formatter before submitting your PRs.

Reply:
> Please make sure to always run the formatter before submitting your PRs.

an you tell more about the formatter , i am beginner and do not know about it , help will be much appreciated :)

Reply:
Workbench formats your snippets every time you run them.
All you have to do is run them once, and then copy the formatted code into `src/Library/demos/Action Bar/code.rs` :)

Reply:
> Workbench formats your snippets every time you run them. All you have to do is run them once, and then copy the formatted code into `src/Library/demos/Action Bar/code.rs` :)

i copied the exact code from the builder after running it , maybe a problem on my side, i will look  into it and manually format it till i figure it out :)

Reply:
[Screen recording](https://github.com/sonnyp/Workbench/assets/30049909/24a2dba3-5afa-476e-9150-0b4805a35e7b)
So you don't get this behavior?

Reply:
> [Screen recording](https://github.com/sonnyp/Workbench/assets/30049909/24a2dba3-5afa-476e-9150-0b4805a35e7b) So you don't get this behavior?

i build the development configuration again hopefully it will be resolved , i remember it formatted my poorly written checkbox code yesterday :)

Reply:
the formatter works correctly now, i formatted the code and thank you very much of reminding me about this feature and for helping me :)

Reply:
Are you sure? Line 10 still looks off.

Reply:
> Are you sure? Line 10 still looks off.

![image](https://github.com/sonnyp/Workbench/assets/104891437/fba12ace-5a2d-421d-bbf6-eee10fc3d674)
Directly copied the formatted code for this made from the workbench :)